go/datas/database.go

```diff
@@ -94,8 +94,13 @@ type Database interface {
 	// Regardless, Datasets() is updated to match backing storage upon return.
 	FastForward(ds Dataset, newHeadRef types.Ref) (Dataset, error)
 
+	// ValidatingBatchStore returns the BatchStore used to read and write
+	// groups of values to the database efficiently. This interface is a low-
+	// level detail of the database that should infrequently be needed by
+	// clients.
+	ValidatingBatchStore() types.BatchStore
```

Comment:
This is very, very, very much intended to remain internal API. Kind of the only reason @rafael-atticlabs and I were OK with having this method on Database is that it was package-internal and only used by `Pull()`. There are some subtle and potentially surprising side effects of calling this method, and I think that exposing it outside the package is asking for long-term pain.

Instead of `cfg.GetDatabase()`, maybe you could use `cfg.GetChunkStore()` and just get/set the Root that way. I know we're talking about "getting rid of" ChunkStore, but that just means we'll replace it with BatchStore throughout the code base.